services/utils.py

```diff
@@ -101,7 +102,9 @@ def get_mirror(status, id, row):
     else:
         host = user_media_url('addons')
 
-    return posixpath.join(host, str(id), row['filename'])
+    url = posixpath.join(host, str(id), row['filename'])
+    params = urllib.urlencode({'filehash': row['hash']})
+    return '{0}?{1}'.format(url, params)
```

Comment:
Is this just for cache busting? Or also for security checks? For cache busting it seems heavy to include the full sha and prefix? It's fine, computers don't mind, but the URL ends up being quite ugly.